test/blocks/commerce/commerce.test.js

```diff
@@ -76,6 +76,27 @@ describe('decorateSearch', () => {
     expect(input).to.exist;
     expect(input.getAttribute('placeholder')).to.equal('Enter offer URL to preview');
   });
+
+  it('calls handle search correctly on keyup', () => {
+    const el = document.createElement('div');
+    decorateSearch(el);
+    const searchWrapper = el.querySelector('.offer-search-wrapper');
+    const offerDetailsWrapper = el.querySelector('.offer-details-wrapper');
+    const input = searchWrapper.querySelector('.offer-search');
+    const e = new KeyboardEvent("keyup", {bubbles : false, cancelable : true, key : "Q", shiftKey : false});
+    Object.defineProperty(e, 'target', {
+      value: {
+        value: 'Q',
+      }
+    });
+    input.dispatchEvent(e);
+    let h4 = el.querySelector('h4');
+    expect(h4).to.equal(null);
+    setTimeout(() => {
```

Comment:
I'm not sure if we should use timeouts in tests. Once might be okay, but if we start adding multiple timeouts, the test suite will take longer than needed to run. Maybe something like [this](https://github.com/adobecom/milo/blob/main/test/blocks/global-navigation/global-navigation.test.js#L158-L161) can help.

Reply:
great input, thank you!